packages/data/abstract/polls_repository/lib/polls_repository.dart

```diff
@@ -20,42 +20,58 @@ abstract class PollsRepository {
 //--------------------------------------------------------------------------
   /// returns arithmetic mean of values of polls
   Map<PollModelType, double> arithmeticMean(int days) {
-    final lengthOfUserPolls = curUserPolls.length;
-
-    /// case lengthOfUserPolls ==  0
-    if (lengthOfUserPolls == 0) return null;
+    //
+    if (curUserPolls.isEmpty) return null;
 
     /// for all other cases
     /// gets a count of poll details
-    final detailsCount = curUserPolls[0].details.length;
-    int mood = 0;
+
+    /// we need to know how many other polls except mood we have
+    /// so we choose any poll and count its details
+    final anyPoll = curUserPolls.values.toList()[0];
```

Comment:
[Itarable.first](https://api.dart.dev/stable/2.10.4/dart-core/Iterable/first.html) might be better candidate instead of `toList()` call. Then convertion to a list does not happen on every call to `arithmeticMean`

Eg `curUserPolls.values.first`.